File: iks-core/src/com/iks/cms/core/sql/query/AbstractChangeQuery.java
package com.iks.cms.core.sql.query;

import com.iks.cms.core.sql.*;

import java.util.*;

/**
 * @author Igor Kaynov
 */
public class AbstractChangeQuery extends SqlQuery {
  protected Table table;
  protected List< UpdateColumn > updateColumns = new ArrayList<>();
  public void addUpdateColumn( Column column, Object value ) {
    updateColumns.add( new UpdateColumn( column, value ) );
  }
  public Table getTable() {
    return table;
  }
  public void setTable( Table table ) {
    this.table = table;
  }
}

File: iks-core/src/com/iks/cms/core/sql/query/AbstractChangeQuery.kt
package com.iks.cms.core.sql.query

import com.iks.cms.core.sql.*

import java.util.*

/**
 * @author Igor Kaynov
 */
open abstract class AbstractChangeQuery(val table: Table) : SqlQuery() {
  var updateColumns: MutableList<UpdateColumn> = ArrayList()
  open fun addUpdateColumn(column: Column, value: Any) {
    updateColumns.add(UpdateColumn(column, value))
  }
}
